effekt/shared/src/main/scala/effekt/Namer.scala

```diff
@@ -695,13 +697,15 @@ object Namer extends Phase[Parsed, NameResolved] {
       tpe
     // TODO reconsider reusing the same set for terms and types...
     case source.BoxedType(tpe, capt) =>
-      BoxedType(resolve(tpe), resolve(capt))
+      BoxedType(resolveBlockType(tpe), resolve(capt))
+    case other => Context.abort(pretty"Expected value type, but got ${other}.")
   }
 
-  def resolve(tpe: source.BlockType)(using Context): BlockType = resolvingType(tpe) {
+  def resolveBlockType(tpe: source.BlockType)(using Context): BlockType = resolvingType(tpe) {
     case t: source.FunctionType  => resolve(t)
     case t: source.BlockTypeTree => t.eff
-    case t: source.BlockTypeRef => resolve(t)
+    case t: source.TypeRef => resolveBlockRef(t)
+    case other => Context.abort(pretty"Expected block type, but got ${other}.")
```

Comment:
TODO: Still need to change these error messages to nicer ones (that's the whole purpose of this exercise!)